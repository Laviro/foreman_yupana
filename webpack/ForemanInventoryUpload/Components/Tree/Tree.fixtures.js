export const props = {
  files: ['file1', 'file2', 'file3'],
};
