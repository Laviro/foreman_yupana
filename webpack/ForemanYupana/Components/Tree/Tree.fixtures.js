export const props = {
  files: { queue: ['file1', 'file2', 'file3'], error: null },
};
