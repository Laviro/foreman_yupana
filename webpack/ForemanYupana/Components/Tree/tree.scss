.queue_error {
  color: #f00;
}

.reports_queue {
  .item {
    margin-bottom: 10px;

    .fa {
      &.fa-download {
        // margin-left: 5px;
        margin-right: 5px;
        cursor: pointer;
      }
    }
  }
}
