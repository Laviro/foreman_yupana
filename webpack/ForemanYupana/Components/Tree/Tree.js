import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Grid, noop } from 'patternfly-react';
import './tree.scss';

const Tree = ({ files: { queue, error }, itemDownload }) => {
  if (error !== null) {
    return (
      <Grid.Col sm={9}>
        <p>Tried to get the uploads queue, but instead got:</p>
        <p className="queue_error">{error}</p>
      </Grid.Col>
    );
  }
  const filesAmount = queue.length;
  const queueTree = queue.map((file, index) => (
    <div className="item" key={index}>
      <Icon name="download" onClick={() => itemDownload(file)} />
      {file}
    </div>
  ));
  return (
    <Grid.Col sm={9}>
      <p>There are currently {filesAmount} report files ready</p>
      <div className="reports_queue">{queueTree}</div>
    </Grid.Col>
  );
};

Tree.propTypes = {
  files: PropTypes.shape({
    queue: PropTypes.arrayOf(PropTypes.string),
    error: PropTypes.string,
  }),
  itemDownload: PropTypes.func,
};

Tree.defaultProps = {
  files: {
    queue: [],
    error: null,
  },
  itemDownload: noop,
};

export default Tree;
