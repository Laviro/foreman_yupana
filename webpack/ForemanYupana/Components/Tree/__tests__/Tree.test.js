import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import Tree from '../Tree';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('Tree', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(Tree, fixtures));
});
