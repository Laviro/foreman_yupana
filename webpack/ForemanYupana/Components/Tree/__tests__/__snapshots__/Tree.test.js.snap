// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree rendering render with props 1`] = `
<Col
  bsClass="col"
  componentClass="div"
  sm={9}
>
  <p>
    There are currently 
    3
     report files ready
  </p>
  <div
    className="reports_queue"
  >
    <div
      className="item"
      key="0"
    >
      <Icon
        name="download"
        onClick={[Function]}
        type="fa"
      />
      file1
    </div>
    <div
      className="item"
      key="1"
    >
      <Icon
        name="download"
        onClick={[Function]}
        type="fa"
      />
      file2
    </div>
    <div
      className="item"
      key="2"
    >
      <Icon
        name="download"
        onClick={[Function]}
        type="fa"
      />
      file3
    </div>
  </div>
</Col>
`;

exports[`Tree rendering render without Props 1`] = `
<Col
  bsClass="col"
  componentClass="div"
  sm={9}
>
  <p>
    There are currently 
    0
     report files ready
  </p>
  <div
    className="reports_queue"
  />
</Col>
`;
